File: medium/1701_average_waiting_time/solution.cpp
#include <vector>

class Solution {
public:
  double averageWaitingTime(std::vector<std::vector<int>> &customers) {
    int currTime = customers[0][0];
    long waitingTime = 0;

    for (const std::vector<int> &customer : customers) {
      int arrival = customer[0];
      int time = customer[1];

      if (arrival > currTime)
        currTime = arrival;

      currTime += time;
      waitingTime += currTime - arrival;
    }

    return (double)waitingTime / customers.size();
  }
};

File: medium/1701_average_waiting_time/solution.py
from typing import List


class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        curr_time = customers[0][0]
        waiting_time = 0

        for arrival, time in customers:
            if curr_time > arrival:
                waiting_time += curr_time - arrival
            else:
                curr_time = arrival

            curr_time += time
            waiting_time += time

        return waiting_time / len(customers)
